feat: wait for actions using `ActionsClient.wait_for`

jooola opened the pull request:
This function allows the users to wait for multiple actions in an efficient way. All actions are queried using a single call, which reduce the potential for running into rate limits.

apricote commented on hcloud/actions/client.py:
```diff
@@ -129,6 +140,115 @@ class ActionsClient(ResourceActionsClient):
+            if wait(self._client._poll_interval_func(retries)):
+                raise ActionGroupException(
+                    [ActionTimeoutException(action=action) for action in running]
+                )
```
Sorry, I was not clear enough. Besides the actions that time out, maybe the user is also interested in the actions that errored, to show a full picture of the issues.

jooola replied:
Python 3.10 EOL is in 3 months (~31 Oct 2026), so I think we can wait until them to:
- Drop support for python 3.10
- Use the new python >= 3.11 Exception Groups: https://peps.python.org/pep-0654/

So we don't have to use this weird `ActionGroupException`, which is once of the reason why this PR has stalled for some time now.

jooola committed:
feat: drop python 3.10

jooola committed:
feat: wait for actions using ActionsClient.wait_for

This function allows the users to wait for multiple actions in an efficient way. All actions are queried using a single call, which reduce the potential for running into rate limits.

In addition, users may also configure a duration based timeout when waiting for actions using:

    action.wait_until_finished(timeout=10) # 10 seconds
    # or
    client.actions.wait_for(..., timeout=10)
